tests/api/cluster/test_bucket.py

```diff
@@ -0,0 +1,23 @@
+from model_mommy import mommy
+import pytest
+
+from controlpanel.api import cluster
+
+
+@pytest.fixture(autouse=True)
+def enable_db_for_all_tests(db):
+    pass
+
+
+@pytest.fixture
+def bucket():
+    return mommy.make('api.S3Bucket', name='test-bucket')
+
+
+def test_arn(bucket):
+    assert cluster.S3Bucket(bucket).arn == 'arn:aws:s3:::test-bucket'
+
+
+def test_create(aws, bucket):
+    cluster.S3Bucket(bucket).create()
+    aws.create_bucket.assert_called_with(bucket.name, False)
```

Comment:
Potentially I'd use the named argument as it helps with booleans and avoid confusion in case we add another argument. (no need to change unless you feel like it, just an observation as tests are also documentation and I had to look at the method definition to remember what was the second argument)

Also, potentially the `aws.create_bucket()` shouldn't care about high-level thing like whether a bucket is "data warehouse" or not, but happy to keep it as it is for the time being.

In my view, as mentioned in another comment, we could change the `aws.create_bucket()` API to receive tags so to be agnostic of how we use S3 buckets, e.g.

```python
aws.create_bucket(bucket_name, {"buckettype": "datawarehouse"})
```

but I do appreciate is a bit long-winded and it would need to transform the Python dictionary into the weird array of dictionaries with `Key`/`Value` internally.

But this is currently used only in `cluster.S3Bucket.create()` which could (should IMHO) pass the `{"buckettype": "datawarehouse"}` tags dictionary (as in my mind the `aws` module is more low-level).

But as I said, no need to change for now.

Reply:
The `create_bucket` function is not called with `is_data_warehouse=False`, and this test fails if we check for that.

As I explained in a previous comment, the `controlpanel.api.aws` module is explicitly wrapping a higher-level interface around the boto3 API, and referring to AP domain objects such as "data warehouse" is deliberate. Passing the tagset to this wrapper layer breaks that level of abstraction. The point of this module is to abstract the use of lower-level AWS API calls to achieve problem domain goals, eg "Create a data warehouse bucket", and not "Create a bucket in S3 with these specific tags".